File: insertion_sort.py
def insertion_sort(list):
	for i in range(1, len(list)):
		current = list[i]
		j = i
		while ((j > 0) and (list[j - 1] > current)):
			list[j] = list[j - 1]
			j = j - 1
		if current != i:
			list[j] = current
	return list

a = [5, 2, 6, 1, 3, 9]
print insertion_sort(a)

File: insertion_sort.js
function insertion_sort(array) {
	for(var i = 0; i < array.length; i++){
		var current = array[i];
		for(var j = i; j > 0 && (array[j - 1] > current); j--){
			array[j] = array[j - 1];
			current = array[j];
		}
		array[j+1] = current;
	}
  return array;
}
